contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,248 @@
+# Federated StatefulSet design
+irfan.rehman@huawei.com, dhilip.kumar.s@huawei.com
+
+# Goal
+
+To outline the use cases for the need of federated Statefulset and determine suitable design/implementation approach for the same.
+
+# Background
+
+The original design proposal of the statefulset in kubernetes, including the use cases and example applications, can be found [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/stateful-apps.md).
+
+
+# Use Cases
+
+1 – A stateful app, for the reasons of high availability, wants the stateful pods distributed in different clusters, such that the set can withstand cluster failures. This represents an app with one single global quorum.
+
+2 – A stateful app, wants replicas distributed in multiple clusters, such that it can form multiple smaller sized local clusters using only local replicas.
+
+# Design Requirements
+
+If we consider the use cases listed above, the main design requirements can roughly be listed as:
+
+1 – An unique, consistent and discoverable identity for each replica/instance across the federated clusters.
```

Comment:
@smartclayton I have mentioned the requirement explicitly as you suggested. I have also dropped the sequentialization requirement, because we are not honoring that as of now.
@chrislovecnm I think @smarterclayton is referring to the quorum application pod instances form after discovering each other.